includes/Admin/Admin_Page.php

```diff
@@ -22,6 +22,14 @@ class Admin_Page {
 	 */
 	protected $admin_ajax;
 
+	/**
+	 * Admin page hook suffix.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	protected $hook_suffix;
```

Comment:
Why introduce a class property for this? This is not used anywhere.

Reply:
You mentioned above that we shouldn't introduce a return value just for a test because it's a bad pattern. I'd argue the same applies here: We shouldn't use a class property to avoid duplicate execution of a method just for a test.

The `add_page()` method should be called only from `add_and_initialize_page()`, it shouldn't be called again in the test.

Reply:
How do we get the value if we don't call the function?

Reply:
We call `add_and_initialize_page()` which calls `add_page()`, so the function will be called. It then sets the `$hook_suffix` class property value and we can get the value from there.

Reply:
You mentioned you wanted developers to be able to get this value. How do you expect them do this?

Reply:
> You mentioned you wanted developers to be able to get this value. How do you expect them do this?

I don't think that's strictly necessary. If we want external developers to use this value, we should probably add a `get_hook_suffix()` public method that returns the value of the class property. That may not be a bad idea, then we can also use it in the test without using reflection.

Reply:
Okay. You want me to add a getter then? Please confirm, I feel we really going around in central and I want to confirm the path forward.

Reply:
Yeah, I think a getter is probably the cleanest way for both the production code and tests.